statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,472 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jul 12 09:35:35 2017
+
+@author: tvzyl
+"""
+import numpy as np
+import pandas as pd
+import warnings
+
+from collections import namedtuple
+from scipy.stats import boxcox
+try:
+    from scipy.special import inv_boxcox
+except ImportError:
+    inv_boxcox = lambda x, lam6da: (x**lam6da-1)/lam6da if lam6da != 0 else np.log(x)
+    
+from scipy.spatial.distance import sqeuclidean
+from scipy.optimize import minimize, basinhopping, brute
+
+Fit = namedtuple("Fit", ["fcast","fitted","alpha", "beta", 
+                         "gamma", "m", "phi", "l0", "b0", 
+                         "s0", "SSE", "level", "slope", 
+                         "season", "opt", "AIC", "BIC", "AICc",
+                         "resid", "k"])
+
+def holt_winters(data, alpha=None, beta=None, gamma=None, m=None, h=1, trend=None, 
+                 damped=False, seasonal=None, phi=None, optimised=True,
+                 boxcoxed=False, remove_bias=False, try_harder=False):  
+    """
+    Holt Winter's Exponential Smoothing
+
+    Parameters
+    ----------
+    data : array-like
+        Time series
+    alpha : float, optional
+        The alpha value of the simple exponnetial smoothing, if the value is
+        set then this value will be used as the value.
+    beta :  float, optional
+        The beta value of the holts trend method, if the value is
+        set then this value will be used as the value.
+    gamma : float, optional
+        The gamma value of the holt winters seasonal method, if the value is
+        set then this value will be used as the value.
+    m : int
+        The number of seasons to consider for the holt winters.
+    h : int
+        The number of time steps to forecast ahead.
+    trend : str {"add", "mul", None}
+        Type of trend component.
+    damped : bool
+        Should the trend component be damped.
+    seasonal : str {"add", "mul", None}
+        Type of seasonal component.
+    phi : float
+        The phi value of the damped method, if the value is
+        set then this value will be used as the value.
+    optimised : bool
+        Should the values that have not beem set above be optimised automatically?
+    boxcoxed : bool
+        Should the boxcox tranform be applied to the data first?
+    remove_bias : bool
+        Should the bias be removed from the fcast and fitted values before being
+        returned?
+    try_harder : bool
+        Should the opptimser try harder to find optimal values?\
+
+    Returns
+    -------
+    results : namedtuple
+        A namedtuple with fcast, fitted and other attributes
+
+    Notes
+    -----
+    This is a full implementation of the holt winters exponential smoothing as 
+    per [1]. This includes all the unsatble methods as well as the stable methods.
+    The implmentaion of the library follows as per the R library as much as possible
```

Comment:
Did you use the R code?  It is not license compatible as far as I know.  Or just look at documentation, which is OK.